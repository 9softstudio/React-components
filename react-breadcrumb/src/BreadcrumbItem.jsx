import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class BreadcrumbItem extends Component {
    constructor(props) {
        super(props);
    }

    static propTypes = {
        label: PropTypes.string,
        isActive: PropTypes.bool,
        onClick: PropTypes.func
    }

    static defaultProps = {
        label: '',
        isActive: false
    }

    render() {
        const { label, isActive, onClick } = this.props;

        return (
            <a className={`breadcrumb-item ${isActive ? 'active' : ''}`} onClick={onClick} href='javascript:;' >{label}</a>
        );
    }
}