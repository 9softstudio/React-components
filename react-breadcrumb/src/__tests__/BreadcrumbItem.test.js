import React from 'react'
import BreadcrumbItem from '../BreadcrumbItem'
import renderer from 'react-test-renderer'

describe('BreadcrumbItem', () => {
    it('isActive = false - component render correctly', () => { 
        const props = {
            isActive : false
        };

        const component = renderer.create(<BreadcrumbItem {...props} />);

        const result = component.toJSON(); 
        
        expect(result).toMatchSnapshot(); 
    });

    it('isActive = true - component render correctly', () => {
        const props = {
            isActive : true
        };

        const component = renderer.create(<BreadcrumbItem {...props} />); 

        const result = component.toJSON(); 

        expect(result).toMatchSnapshot(); 
    });
})