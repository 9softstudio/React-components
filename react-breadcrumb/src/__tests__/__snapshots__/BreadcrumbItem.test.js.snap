// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItem isActive = false - component render correctly 1`] = `
<a
  className="breadcrumb-item "
  href="javascript:;"
  onClick={undefined}
>
  
</a>
`;

exports[`BreadcrumbItem isActive = true - component render correctly 1`] = `
<a
  className="breadcrumb-item active"
  href="javascript:;"
  onClick={undefined}
>
  
</a>
`;
